bigbite/themer
# [BBT-58] Validate theme.json is loaded

## Description
[BBT-58](https://b5ecom.atlassian.net/browse/BBT-58)

Checks if a theme.json file is readable and can be decoded upon the themer react app being loaded. Error messages will be sent through to be displayed in themer.

I have currently stopped the fields being rendered if the theme.json is invalid as it looks like these settings are generated based on the theme.json file as a invalid theme.json break the react app from the fields. This will most likely not need to stopped from rendering when the fields are generated from the schema or another source.

Wordpress does not give any errors for invalid theme.json it just builds out an array from what it has available. Because of this the only ways I can really validate we have a valid theme.json loaded is by checking if the file is readable and if it can be decoded.

I think validating our fields and what we are saving will be our main point of validation and this is a nice to have. If the theme.json is invalid then this is on the theme developer, just like having an invalid header.php you will get a PHP warning/error.

I've just used the default wordpress notice component for displaying errors for now not sure if we want to change any styling to this @jonnywatersbb 


## Change Log
- Adds endpoint for validating theme.json is loaded.
- Added a simple notice component for error messages.

## Steps to test
1. Load a valid theme.json and see standard behaviour.
2. Load theme.json with invalid json see an error message in themer
3. Rename theme.json to anything so it can't be loaded and see a error message in themer.

## Screenshots/Videos
http://bigbite.im/v/dOsqMu

## Checklist:


[BBT-58]: https://b5ecom.atlassian.net/browse/BBT-58?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
